isisd/isis_cli.c

```diff
@@ -150,23 +169,44 @@ DEFPY_YANG(ip_router_isis, ip_router_isis_cmd, "ip router isis WORD$tag",
 		return CMD_SUCCESS;
 	}
 	ifp = nb_running_get_entry(NULL, VTY_CURR_XPATH, false);
-	if (ifp)
-		area = isis_area_lookup(tag, ifp->vrf_id);
 
+	if (!vrf_name && ifp->vrf_id == VRF_DEFAULT)
+		vrf_name = VRF_DEFAULT_NAME;
+
+	if (ifp->vrf_id != VRF_DEFAULT) {
+		vrf = vrf_lookup_by_id(ifp->vrf_id);
+		if (vrf) {
+			if (!vrf_name)
+				vrf_name = vrf->name;
+
+			if (strcmp(vrf->name, vrf_name) != 0) {
+				vty_out(vty, "%% interface %s not in %s vrf \n", ifp->name, vrf_name);
```

Comment:
It would be better to perform this validation check in the appropriate northbound callback and not in this CLI wrapper (so that it applies to all northbound clients).

Reply:
This validation check is moved to northbound callback.